roles/couchbase-infra/tasks/main.yml

```diff
@@ -1,121 +1,198 @@
-  #       I can't find a way to check label_selectors based on an OR of the key, and the 
-  #       field_selectors I might match on are not supported. It's also not supported to
-  #       loop over a block with an until, so that solution won't work. I also can't find
-  #       a way to inject a new label to use into the operator pod as it is being created
-  #       by the subscription. ¯\_(ツ)_/¯
-  #k8s_facts:
-  #  kind: ClusterServiceVersion
-  #  api_version: v1
-  #  namespace: "{{ operator_namespace }}"
-  #  label_selectors:
-  #    - app = couchbase-operator
-  #    - name = couchbase-operator
-  #register: cb_op_pod
-  #until: "'Running' in (cb_op_pod | json_query('resources[].status.phase'))"
-  #retries: 10
-  #delay: 10
-  pause:
-    seconds: 100
-
-# For BZ 1678285
-# https://bugzilla.redhat.com/show_bug.cgi?id=1678285
-- name: Apply RBAC for Couchbase Operator
-  k8s:
-    definition: "{{ lookup('template', 'couchbase-operator-rbac.yaml.j2') }}"
-
-- name: Create the Couchbase Auth Secret
-  k8s:
-    definition:
-      apiVersion: v1
-      kind: Secret
-      metadata:
-        name: cb-cluster-auth
-        namespace: "{{ operator_namespace }}"
-      type: Opaque
-      data:
-        username: QWRtaW5pc3RyYXRvcg==
-        password: cGFzc3dvcmQ=
-
-# Pre-pull the image to avoid CB operator timeout
-- name: Pre-Pull Couchbase Container Image
-  k8s:
-    definition:
+# Plays common to the Deploy and Destroy blocks are first
+#- name: My Task
+#    k8s:
+#      src:
+
+
+# Deploy block is run as long as the finalizer has not set teardown=true
+- name: Deploy
+  when: teardown is undefined
+  block:
+
+  - name: Deploy couchbase operator
+    k8s:
+      definition: "{{ lookup('template', 'couchbase-operator.yaml.j2') }}"
+
+  - name: Get couchbase operator CSV
+    k8s_facts:
+      api_version: operators.coreos.com/v1alpha1
+      kind: ClusterServiceVersion
+      namespace: "{{ operator_namespace }}"
+    register: csvs
+    retries: 10
+    delay: 5
+
+  - name: Wait for Couchbase Operator to be Running....
+    #FIXME: Some versions of the operator use an "app" label and others use a "name" label.
+    #       I can't find a way to check label_selectors based on an OR of the key, and the 
+    #       field_selectors I might match on are not supported. It's also not supported to
+    #       loop over a block with an until, so that solution won't work. I also can't find
+    #       a way to inject a new label to use into the operator pod as it is being created
+    #       by the subscription. ¯\_(ツ)_/¯
+    #k8s_facts:
+    #  api_version: operators.coreos.com/v1alpha1
+    #  kind: ClusterServiceVersion
+    #  namespace: "{{ operator_namespace }}"
+    #  name: "{{ item.metadata.name }}"
+    #register: cb_op_pod
+    #until: "'Running' in (cb_op_pod | json_query('resources[].status.phase'))"
+    #with_items: "{{ csvs.resources }}"
+    #retries: 10
+    #delay: 10
+    pause:
+      seconds: 100
```

Comment:
Instead of a fixed amount of time, should we query and have a retry w/ delay?

Reply:
Yes, but there is the problem with the above commented-out block, so the sleep was a stop-gap. I haven't come up with a way to test for "upness" that is consistent between k8s/OCP3 and OCP4 because of a difference in metadata.